metrics/release_suite_test.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+type inputHeader struct {
+	Name string
+	Help string
+}
+
+func TestMetricsRelease(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Metrics Release Test Suite")
+}
+
+var (
+	testEnv *envtest.Environment
+	ctx     context.Context
+	cancel  context.CancelFunc
+)
+
+var _ = BeforeSuite(func() {
+
+	ctx, cancel = context.WithCancel(context.TODO())
+	testEnv = &envtest.Environment{}
+	cfg, err := testEnv.Start()
+	Expect(err).NotTo(HaveOccurred())
+	Expect(cfg).NotTo(BeNil())
+
+	k8sManager, _ := ctrl.NewManager(cfg, ctrl.Options{
+		MetricsBindAddress: ":8081",
+		LeaderElection:     false,
+	})
+
+	go func() {
+		defer GinkgoRecover()
+		Expect(k8sManager.Start(ctx)).To(Succeed())
+	}()
+})
+
+var _ = AfterSuite(func() {
+	cancel()
+	By("tearing down the test environment")
+	err := testEnv.Stop()
+	Expect(err).NotTo(HaveOccurred())
+})
+
+func createNewCounterReader(histHeader inputHeader, labels string, renderHeader bool, count int) string {
+	metadata := "# HELP %s %s\n# TYPE %s counter\n" // metadata is required to query
+	totalData := "%s{%s} %d\n"
+	readerData := ""
+
+	if renderHeader {
+		readerData = fmt.Sprintf(metadata, histHeader.Name, histHeader.Help, histHeader.Name)
+	}
+	readerData += fmt.Sprintf(totalData, histHeader.Name, labels, count)
+
+	return readerData
+}
+
+func createNewHistogramReader(histHeader inputHeader, timeBuckets []string, bucketsData []int, labels string, sum float64, count int) string {
```

Comment:
You can drop the `hist` prefix here and in the variables as this function is referencing histograms already. Also, the name of the function should be `createHistogramReader`. It's frequent to see `New` in Go functions when they limit themselves to create an instance of a struct.

Reply:
done